backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py

```diff
@@ -38,3 +30,62 @@ def get_package_info(name: str, version: str) -> dict:
         else:
             LOG.error("Unable to find package info for %s: HTTP %s", name, r.status_code)
             return {}
+
+
+def get_package_license(package_info: dict, package_name: str) -> list[str]:
+    # Official Spec: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#license
+    # We support the official spec, as well as the "deprecated" syntax with objects/arrays in a
+    # `license` or `licenses` property
+    #
+    # So, we should be able to parse all of these:
+    # license: "GPL-3.0"
+    # license: "(GPL-3.0 OR MIT)" # SPDX expression. We will return the expression whole as a string
+    # license: [
+    #     "GPL-3.0",
+    #     "MIT"
+    # ]
+    # license: {
+    #     type: "GPL-3.0",
+    #     url: "..."
+    # }
+    # licenses: [
+    #     { type: "GPL-3.0", url: "..." },
+    #     { type: "MIT", url: "..." }
+    # ]
+
+    if "license" in package_info:
+        license = package_info["license"]
+    elif "licenses" in package_info:
+        license = package_info["licenses"]
+    else:
+        LOG.debug('No license information in package info for "%s"', package_name)
```

Comment:
For reviewer, should this be debug? We log as "Error" on L31 when we can't find the package info.

Reply:
I think the information logged here should be a `warning` . But it also depends on how common this case is. 
If its really common, it makes sense to have it a `debug` log. If its not, then it should be a `warning`